client/app/services/emergencyNumber.js

```diff
@@ -0,0 +1,181 @@
+angular.module('distress')
+.factory('EmergencyNumber', function(){
+  var instance = {};
+
+  instance.getIt = function(coords, callback){
+    var location = new google.maps.LatLng(coords.latitude, coords.longitude);
+    var request = {location: location, radius: 1000};
+
+    //The PlacesService method requires a DOM node for exactly that purpose.
+    //That's what #myDiv is for.
+    var service = new google.maps.places.PlacesService(document.getElementById('myDiv'));
+
+    // Initial nearbySearch call returns object contatining place_id
+    // required parameter for nested getDetails call
+    service.nearbySearch(request, function(results, status){
+      if (status === google.maps.places.PlacesServiceStatus.OK) {
+        var placeId = { placeId: results[0].place_id };
+
+        service.getDetails(placeId, function(results, status){
+          if (status === google.maps.places.PlacesServiceStatus.OK) {
+            var country = results.formatted_address.split(',').slice(-1)[0].trim();
+            var emergencyNumber = countryNumbers[country];
+            callback(emergencyNumber);
+          }
+        });
+      }
+    });
+  };
+
+  //still need to check names
+  var countryNumbers = {
```

Comment:
That would work. So the tradeoff is another dependency to inject but centralizing our resource. Do you think there will be any other controller that will need access to this in the future?